src/rtc.c

```diff
@@ -346,17 +345,38 @@ void RTC_init(hourFormat_t format, sourceClock_t source, bool reset)
   RtcHandle.Init.OutPutType = RTC_OUTPUT_TYPE_OPENDRAIN;
 #endif /* STM32F1xx */
 
+  /* Ensure backup domain is enabled before we init the RTC so we can use the backup registers for date retention on stm32f1xx baords */
+  enableBackupDomain();
+
   HAL_RTC_Init(&RtcHandle);
 
+#if defined(STM32F1xx)
+  // Copy date data back out of the BackUp registers
+  RTC_DateTypeDef BackupDate;
+  RTC_TimeTypeDef DummyTime;
+  uint32_t dateMem;
+  dateMem = HAL_RTCEx_BKUPRead(&RtcHandle, RTC_BKP_DR3);
```

Comment:
could the Backup register be defined, like the following
#define RTC_BKP_SAVE_THE_DATE    RTC_BKP_DR2
and the accessed by RTC_BKP_SAVE_THE_DATE, RTC_BKP_SAVE_THE_DATE+1

Reply:
Just a quick note, pay attention to not use the same BKP register used by the core for BL management:
https://github.com/stm32duino/Arduino_Core_STM32/blob/f4c200f9d7d1258a15e1a2c56dbdd9d370e1b5b9/cores/arduino/stm32/backup.h#L41-L60

Reply:
I have added in a backup register definition as suggested. 
The BL looks to be using DR4 and the rtc is using DR2 and DR3 so there shouldn't be any conflict there.